Shared/ViewModels/ItemViewModel/SeasonItemViewModel.swift

```diff
@@ -38,56 +39,51 @@ final class SeasonItemViewModel: ItemViewModel, EpisodesRowManager {
 			return L10n.unaired
 		}
 
-		if item.missing {
-			return L10n.missing
-		}
-
 		guard let playButtonItem = playButtonItem, let episodeLocator = playButtonItem.getEpisodeLocator() else { return L10n.play }
 		return episodeLocator
 	}
 
-//	private func requestEpisodes() {
-//		LogManager.shared.log
-//			.debug("Getting episodes in season \(item.id!) (\(item.name!)) of show \(item.seriesId!) (\(item.seriesName!))")
-//		TvShowsAPI.getEpisodes(seriesId: item.seriesId ?? "", userId: SessionManager.main.currentLogin.user.id,
-//		                       fields: [.primaryImageAspectRatio, .seriesPrimaryImage, .seasonUserData, .overview, .genres, .people],
-//		                       seasonId: item.id ?? "")
-//			.trackActivity(loading)
-//			.sink(receiveCompletion: { [weak self] completion in
-//				self?.handleAPIRequestError(completion: completion)
-//			}, receiveValue: { [weak self] response in
-//				self?.episodes = response.items ?? []
-//				LogManager.shared.log.debug("Retrieved \(String(self?.episodes.count ?? 0)) episodes")
-//
-//				self?.setNextUpInSeason()
-//			})
-//			.store(in: &cancellables)
-//	}
+	private func requestEpisodes() {
+		LogManager.shared.log
```

Comment:
So, I actually found out that we shouldn't necessarily do it this way. I had the thought "what would happen if there were hundreds of episodes" (which happens) and looked to the web and found out that they use the `NextUp` endpoint and pass in the series ID. Sorry to say that I will initially reject this implementation for that but it is a simple endpoint and will end up being cleaner here.

Reply:
re-review request changes comment

Reply:
I fiddled around and conclude that `NextUp` doesn't work with a specific season id[1]. Namely, it only tells you nextup in the enclosing series but not nextup in a specific season. So right now I still leverage `NextUp`, but validate if the returning nextup items belong to the current season. If not, falling back to the old mechanism a.k.a picking the first unplayed episode as nextup.

[1]: Including providing season id to the `seriesId` argument, populating the `parentId` argument with either a season or series id.

Reply:
> you can also fix this for the series as well if you want to. I can't remember if it works and if it does that it uses the same endpoint.

I think the series view model has been using the `NextUp` endpoint.